### PR TITLE
Add fullscreen image preview

### PR DESCRIPTION
**Pull request by MohamedAli1937**
## Description

Add a simple fullscreen image preview feature allowing users to view the generated image before downloading.

## Type of Change

- [ ] Bug fix
- [x] New feature
- [ ] Breaking change
- [ ] Documentation update

## How Has This Been Tested?

Describe the tests you ran to verify your changes.

- [x] Existing tests pass
- [ ] New tests added
- [x] Manual testing

## Screenshots (if app

https://github.com/user-attachments/assets/947da446-dc6d-48a8-8d46-82c2dc754803

licable)

## Checklist

- [x] My code follows the project's style guidelines
- [x] I have performed a self-review
- [x] I have added/updated documentation as needed
- [x] My changes generate no new warnings
- [x] Tests pass locally

### CODE REVIEW

**Review by ivantha — COMMENTED** [review]
## ◎ Argus Review

This PR introduces a clean, minimal fullscreen image preview feature with good separation of concerns and sensible UX touches (Escape-key support, CSS fade-in, drag/select prevention). The implementation is readable and integrates naturally into the existing `PreviewPane` structure. However, there are notable concerns around accessibility (missing ARIA attributes and focus management), a subtle `useEffect` dependency bug caused by an unstable `onClose` reference, and a missing format-validation guard in the data URI construction. No automated tests accompany the change.

File: imagelab-frontend/src/components/Preview/PreviewPane.tsx [hunk 1]
```diff
@@ -39,14 +50,16 @@ export default function PreviewPane() {
     usePipelineStore();
   const [originalZoom, setOriginalZoom] = useState<number | null>(null);
   const [processedZoom, setProcessedZoom] = useState<number | null>(null);
```
Comment (ivantha) [inline imagelab-frontend/src/components/Preview/PreviewPane.tsx]: **[INFO]** Positive: Typed state with discriminated shape cleanly models open/closed states

The `fullscreenImage` state is typed as `{ str: string; format: string } | null`, which cleanly represents both the "open" (with data) and "closed" (`null`) states in a single value. Using `null` as the closed sentinel avoids an extra `isOpen` boolean and ensures that `imageStr` and `format` are always co-located. This is a well-applied TypeScript discriminated union pattern.

**Suggestion:**
```
No change needed. This is a good pattern — continue applying it for similar feature states in the codebase.
```

**Review by ivantha — COMMENTED** [review]
## ◎ Argus Review

This PR adds a well-conceived fullscreen image preview feature with thoughtful UX details (scroll lock, focus management, loading spinner, Escape key support, and body scroll restoration). However, it contains two **critical React correctness bugs** in `ImageFullscreenPreview.tsx`: hooks are declared after a conditional early return (Rules of Hooks violation), and `onClose()` is called directly during the render phase (a side effect in render). The `ALLOWED_FORMATS` constant is also duplicated across three files creating a maintenance hazard. A high-severity infrastructure change (global test environment `node` → `jsdom`) is made without documentation or verification that existing tests are unaffected. The test suite validates the incorrect render-phase side-effect behaviour, meaning the tests must also be updated when the bugs are fixed.

**Suggested Version Bump:** `minor`

### ⚠️ Breaking Changes

- **[INFO]** No Breaking Changes to Public API: **Previous behavior**: `ZoomControls` was an internal component with `disabled`, `onZoomIn`, `onZoomOut` props.

**New behavior**: `ZoomControls` now requires an additional `onFullscreen` prop.

**Who is affected**: `ZoomControls` is not exported and is only consumed within `PreviewPane.tsx`. All call sites have been updated in this PR.

**Migration path**: N/A — internal component change only.

All other changes are additive (new component, new test file, new optional store behaviour). No public APIs were removed or incompatibly modified.
  > **Suggestion:** No action needed. This is a minor/additive change.


### PR Description Issues

- **[HIGH]** PR Description Contains Unfilled Template Text in Testing Section: The testing section contains the raw template placeholder text: *"Describe the tests you ran to verify your changes."* This was not replaced with actual manual testing details (e.g., what browsers were tested, what scenarios were exercised, what edge cases were checked).
  > **Suggestion:** Replace the template text with concrete testing details:

> **Manual Testing:**
> - Opened the app in Chrome/Firefox and clicked the fullscreen button for both original and processed images
> - Verified the Escape key closes the modal
> - Verified clicking outside the image (backdrop) closes the modal
> - Verified clicking the image itself does not close the modal
> - Confirmed the loading spinner disappears once the image loads
> - Tested with PNG, JPEG, and WebP formats
- **[MEDIUM]** `New tests added` Checkbox Unchecked Despite New Test File Being Added: The PR description shows `- [ ] New tests added` (unchecked), but the diff clearly includes a new test file `ImageFullscreenPreview.test.tsx` with 85 lines of tests. This is a direct mismatch between the checklist and the actual changes.
  > **Suggestion:** Update the checklist to reflect the reality:
```
- [x] New tests added
```
- **[LOW]** Screenshots Section Heading Is Truncated: The screenshots section heading reads `## Screenshots (if app` — the word `applicable)` was cut off mid-word, indicating a copy-paste or editing error in the PR template.
  > **Suggestion:** Fix the heading:
```
## Screenshots (if applicable)
```
- **[LOW]** Description Omits Mention of Test Infrastructure Changes: The PR changes the global Vitest environment from `node` to `jsdom`, adds four new testing dependencies to `package.json`, and creates a `vitest.setup.ts` file. None of these changes are mentioned in the PR description. These are non-trivial infrastructure changes that reviewers and maintainers should be explicitly made aware of.
  > **Suggestion:** Add a section or bullet point noting the test infrastructure changes:

> **Test Infrastructure:**
> - Added `@testing-library/react`, `@testing-library/jest-dom`, `@testing-library/user-event`, and `jsdom` as dev dependencies to support React component testing
> - Changed global Vitest environment from `node` to `jsdom` to enable component tests without per-file overrides
> - Added `vitest.setup.ts` for global jest-dom matchers

**Comment by ivantha** [comment]
Thanks for the PR! The 'Frontend Lint' check is failing. Could you run `npm run lint` locally and push the fixes? Happy to take another look once CI is green.

**Comment by MohamedAli1937** [comment]
@ivantha Squashed the commits into a clean history and force-pushed the branch. PR is updated and ready for review 👍

**Comment by ivantha** [comment]
Thanks @MohamedAli1937! A fullscreen preview modal was merged as #163 (2026-03-12). Closing this as the feature is live. If you have improvements to the existing preview (e.g. zoom, download from modal), please open a focused follow-up PR!
